backend/uclapi/dashboard/app_helpers.py

```diff
@@ -30,3 +32,18 @@ def generate_app_id():
     final = "A" + key
 
     return final
+
+
+def generate_app_client_id():
+    sr = SystemRandom()
+
+    client_id = ''.join(str(sr.randint(0, 9)) for _ in range(16))
+    client_id += "."
+    client_id += ''.join(str(sr.randint(0, 9)) for _ in range(16))
```

Comment:
could you use f-strings or .format() syntax here? It's slightly more pythonic.

Reply:
really? Everywhere I've read online says that the ''.join() is more Pythonic. @Faiz?

Reply:
My comment here wasn't about `join` and more about the `+=` concatenation

that could be

```
client_id = '{}.{}'.format(join(str(sr.randint(0, 9)) for _ in range(16)), join(str(sr.randint(0, 9)) for _ in range(16)))
```
not sure what the best f-string syntax would look like cc @jermenkoo

Reply:
Whilst your version does look more pythonic, it's far harder to understand and manage. I'm struggling to follow it.

Will consider revising, but we need something that's not about four brackets deep :P

Reply:
```
client_id = '{}.{}'.format(
    ''.join(str(sr.randint(0, 9)) for _ in range(16)),
    ''.join(str(sr.randint(0, 9)) for _ in range(16))
)
```

Reply:
Still about four brackets deep but at least with that indentation it's understandable so I'll accept that change :)

Reply:
sorry late comer. 

Since we are using a function, I would cast the ''.join to a function variable and then call it twice for readability. 

`fun = lambda  : ''.join(str(sr.randint(0, 9)) for _ in range(16))

client_id = '{}.{}'.format(fun(), fun())`

Reply:
Named lambdas are pretty much frowned upon, can't you just extract it somewhere, e.g. `utils` as `generate_client_id()` and use that?

Not coupled, testable, replaces 4 lines with 1.

Reply:
Closing this as outdated now: I went with Wilhelm's `.format` syntax. Been using a lot of them at work and it's a nice way of doing things :)